src/index.ts

```diff
@@ -150,57 +168,78 @@ async function clientAutoMode(r: NginxHTTPRequest): Promise<void> {
 
     try {
       fs.mkdirSync(challengePath, { recursive: true })
     } catch (e) {
-      log.error(
-        `Error creating directory to store challenges. Ensure the ${challengePath} directory is writable by the nginx user.`
-      )
-
-      return r.return(500, 'Cannot create challenge directory')
+      retVal.info.error = `Error creating directory to store challenges. Ensure the ${challengePath} directory is writable by the nginx user.`
+      return retVal
     }
 
     certificatePem = await client.auto({
       csr: Buffer.from(csr.pkcs10Ber),
       email,
       termsOfServiceAgreed: true,
-      challengeCreateFn: async (authz, challenge, keyAuthorization) => {
-        log.info('Challenge Create', { authz, challenge, keyAuthorization })
+      challengeCreateFn: async (_, challenge, keyAuthorization) => {
         log.info(
           `Writing challenge file so nginx can serve it via .well-known/acme-challenge/${challenge.token}`
         )
-        ngx.log(
-          ngx.INFO,
-          `njs-acme: [auto] Writing challenge file so nginx can serve it via ${challengePath}/${challenge.token}`
-        )
         const path = joinPaths(challengePath, challenge.token)
         fs.writeFileSync(path, keyAuthorization)
       },
       challengeRemoveFn: async (_authz, challenge, _keyAuthorization) => {
         const path = joinPaths(challengePath, challenge.token)
         try {
           fs.unlinkSync(path)
-          log.info(`removed challenge ${path}`)
+          log.info(`Removed challenge ${path}`)
         } catch (e) {
-          log.error(`failed to remove challenge ${path}`)
+          log.error(`Failed to remove challenge ${path}`)
         }
       },
     })
     certInfo = await readCertificateInfo(certificatePem)
     fs.writeFileSync(certPath, certificatePem)
-    log.info(`wrote certificate to ${certPath}`)
+    log.info(`Wrote certificate to ${certPath}`)
   }
 
-  const info = {
-    certificate: certInfo,
-    renewedCertificate: renewCertificate,
-  }
+  retVal.success = true
+  retVal.info.certificate = certInfo
+  retVal.info.renewedCertificate = renewCertificate.toString()
+
+  return retVal
+}
+
+/**
+ * Using AcmeClient to create a new account. It creates an account key if it doesn't exist
+ * @param {NginxHTTPRequest} r Incoming request
+ * @returns void
+ */
+async function clientNewAccount(r: NginxHTTPRequest): Promise<void> {
+  const accountKey = await readOrCreateAccountKey(acmeAccountPrivateJWKPath(r))
+  // Create a new ACME account
+  const client = new AcmeClient({
+    directoryUrl: acmeDirectoryURI(r),
+    accountKey: accountKey,
+  })
+  // display more logs
+  client.api.minLevel = LogLevel.Debug
+  // conditionally validate ACME provider cert
+  client.api.setVerify(acmeVerifyProviderHTTPS(r))
 
-  return r.return(200, JSON.stringify(info))
+  try {
+    const account = await client.createAccount({
+      termsOfServiceAgreed: true,
+      contact: ['mailto:test@example.com'],
```

Comment:
Was this intentionally left in?  Or do we need to ask the user to provide contact information as configuration?  Suspect this might be quirk my maybe it bears clarification in the code via a comment if that is the case.

Reply:
Maxim originally wrote this as an ACME client library for NJS, with the idea that implementations would include it. He added the initial reference implementation that I helped to refine. So this function is leftovers from that library-first idea, that someone may want to make a different implementation that may need to create an account as a separate step. I wonder if we can reorganize things to make this more clear?

Reply:
I've moved those methods to `examples.ts` with some header explanation. Thanks!